src/Button.cc

```
@@ -441,13 +447,62 @@ void Button::setPenWidth(QPainter *painter, const qreal scale)
+{
+    if (m_dpr > 0.0 && m_invertible) {
+        const QPointF devInd = m_trans.map(p);
+        const QPointF phys(devInd.x() * m_dpr, devInd.y() * m_dpr);
+        const QPointF physSnapped(qRound(phys.x()), qRound(phys.y()));
+        const QPointF devIndSnapped(physSnapped.x() / m_dpr, physSnapped.y() / m_dpr);
+        return m_inv.map(devIndSnapped);
+    }
+    return p;
+}
+
+qreal PixelSnapper::snap(const qreal v) const
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)